src/oic/oauth2/message.py

```diff
@@ -439,6 +445,15 @@ def _add_value(self, skey, vtyp, key, val, _deser, null_allowed):
                         val = _deser(val, sformat="dict")
                     except Exception as exc:
                         raise DecodeError(ERRTXT % (key, exc))
+                elif vtyp == int:
+                    try:
+                        self._dict[skey] = int(val)
+                    except (ValueError, TypeError):
```

Comment:
I don't see why we should ignore `TypeError` here? If your only wish is to pass a message with more information you can do `except (ValueError, TypeError) as exc:  raise exc(msg)`.